runtime/vm/exceptiondescribe.c

```diff
@@ -229,40 +229,54 @@ printStackTraceEntry(J9VMThread * vmThread, void * voidUserData, UDATA bytecodeO
  * @param callback The callback function to be invoked for each stack frame.
  * @param userData Opaque data pointer passed to the callback function.
  * @param pruneConstructors Non-zero if constructors should be pruned from the stack trace.
+ * @param sizeOfWalkstateCache Non-zero if exception is walkstate cache instead of exception object.
+ * 								Indicatest the size of cache.
  * @return The number of times the callback function was invoked.
  *
  * @note Assumes VM access
  **/
 UDATA
-iterateStackTrace(J9VMThread * vmThread, j9object_t* exception, callback_func_t callback, void * userData, UDATA pruneConstructors, UDATA skipHiddenFrames)
+iterateStackTraceImpl(J9VMThread * vmThread, j9object_t* exception, callback_func_t callback, void * userData, UDATA pruneConstructors, UDATA skipHiddenFrames, UDATA sizeOfWalkstateCache, BOOLEAN walkBackIsException)
 {
 	J9JavaVM * vm = vmThread->javaVM;
 	UDATA totalEntries = 0;
-	j9object_t walkback = J9VMJAVALANGTHROWABLE_WALKBACK(vmThread, (*exception));
+	void *walkback = NULL;
+
+	if (walkBackIsException) {
```

Comment:
This flag name is very confusing. Something like `exceptionIsWalkback` or the opposite `exceptionIsJavaObject` would be much clearer.